### ISSUE TITLE
dev-python/yara-python suffers undefined symbol: yr_finalize

### ISSUE THREAD

**Issue by ikelos** [pentoo/pentoo-overlay#397]
I believe both 3.7.0 and 3.8.1 of yara-python suffer the issue of not building correctly.  The compilation completes, but attempting to import it in python gives the undefined symbol error.

The reasons are documents in [yara-python issue 7](https://github.com/VirusTotal/yara-python/issues/7)

Basically, a copy of the yara source needs to be present under the yara directory, but is not shipped with the yara-python source package.  I'll attach an ebuild that resolves the issue...

**Comment by ikelos** [comment]
[yara-python-3.8.1-r1.ebuild.txt](https://github.com/pentoo/pentoo-overlay/files/2628901/yara-python-3.8.1-r1.ebuild.txt)

**Comment by ikelos** [comment]
It also includes python3.7 as a target.

**Comment by blshkv** [comment]
Thanks! I'm surprised that you don't have access and can't submit it yourselves ;-)

**Comment by ikelos** [comment]
Ah, I may well have, I don't recall, sorry!  5:S  I'll check it next time before I file another bug.  Thanks!  5:)

**Comment by ikelos** [comment]
Looks like this got reverted in yara-3.10.0, and I checked but I don't have commit rights...  5;)

**Comment by blshkv** [comment]
I just pushed a simple patch to compile against a system library. We control it and can make sure that the correct version of library is installed before we compile a python module.

**Comment by ikelos** [comment]
Awesome, nice fix (and speedily applied!).  Thanks!  5:)  Dunno why the yara-python guys didn't have it rigged like that in the first place?  5:S

**Comment by blshkv** [comment]
They recommend to pull the latest git and compile. That will break if the API changed. So the usual Gentoo approach is better

**Comment by blshkv** [comment]
not sure if it was there before, but I found an option to build with shared library:
```setup.py build --dynamic-linking```

I'm changing our build to use that official option

**Comment by ikelos** [comment]
Ok, think I fixed it.  The `--dynamic-linking` option was being used to build the library within the source directory, but `distutils-r1_python_compile` then compiles it again and puts it in a build directory, and this is the one the eclass installs.  The solution was to pass the option directly in to `distutils-r1_python_compile`.

I've also added a note about the dex flag because we technically don't turn it on (`setup.py` would throw an error if we tried to pass both options).  I don't know whether the feature is enabled because it was enabled in the shared library or not.  

I've applied it as a bump (because the earlier version could be installed without problem though it was broken), but I haven't removed the old versions or made a bump for 4.0.0, so I'll leave this open until we've cleaned it up to the point where everybody's happy with it.  5:)
